spec.bs

```diff
@@ -785,57 +877,87 @@ Issue: This would ideally be replaced by a more descriptive algorithm in Infra.
-   |auctionSignals|, |perBuyerSignals|, |sellerSignals|, |browserSignals| ».
-1. TODO: Set URLs passed to sendReportTo() into the FencedFrameConfig so they can be fetched when
-  ad renders.
-
+<div algorithm>
+To <dfn>report win</dfn> given a [=leading bid info=] |leadingBidInfo|, a [=string=]
+|sellerSignals| and an [=ordered map=] |browserSignals| whose [=map/keys=] are [=strings=] and whose
+[=map/values=] are {{any}}:
+
+  1. Let |config| be |leadingBidInfo|'s [=leading bid info/auction config=].
+  1. Let |winner| be |leadingBidInfo|'s [=leading bid info/leading bid=].
+  1. Let |perBuyerSignals| be |config|'s [=auction config/per buyer signals=].
+  1. Let |buyer| be |winner|'s [=generated bid/interest group=]'s [=interest group/owner=].
+  1. Let |perBuyerSignalsForBuyer| be |perBuyerSignals|[|buyer|] if that member [=map/exists=],
+    and null otherwise.
+  1. [=map/Remove=] |browserSignals|["`desirability`"].
+  1. TODO: [=map/Set=] |browserSignals|["`interestGroupName`"] to |winner|'s [=interest group/name=]
+    if the tuple of interest group owner, name, bidding script URL and ad creative URL were jointly
+    k-anonymous.
+  1. TODO: [=map/Set=] |browserSignals|["madeHighestScoringOtherBid"] are based on the auction the interest group was
+    directly part of. If that was a component auction, they're from the component auction. If that was
+    the top-level auction, then they're from the top-level auction. Component bidders do not get these
+    signals from top-level auctions since it is the auction seller joining the top-level auction,
+    instead of winning component bidders joining the top-level auction directly.
+  1. If |leadingBidInfo|'s [=leading bid info/bidding data version=] is not null:
+    1. [=map/Set=] |browserSignals|["dataVersion"] to |leadingBidInfo|'s
+      [=leading bid info/bidding data version=].
+  1. [=map/Set=] |browserSignals|["adCost"] to |winner|'s [=generated bid/ad cost=].
+  1. [=map/Set=] |browserSignals|["seller"] to |config|'s [=auction config/seller=].
+  1. If |leadingBidInfo|'s [=leading bid info/top level seller=] is not null:
+    1. [=map/Set=] |browserSignals|["topLevelSeller"] to |leadingBidInfo|'s
+      [=leading bid info/top level seller=].
+  1. If |winner|'s [=generated bid/modeling signals=] is not null:
+    1. [=map/Set=] |browserSignals|["modelingSignals"] to |winner|'s
+      [=generated bid/modeling signals=].
+  1. Let |buyerReportingScript| be the result of [=fetching script=] with |winner|'s
+    [=generated bid/interest group=]'s [=interest group/bidding url=].
+  1. Run [=evaluating a reporting script=] with |buyerReportingScript|, "reportWin", and «
+    |leadingBidInfo|'s [=leading bid info/auction config=]'s [=auction config/auction signals=],
+    |perBuyerSignalsForBuyer|, |sellerSignals|, |browserSignals| ».
+  1. TODO: Set URLs passed to sendReportTo() into the FencedFrameConfig so they can be fetched when
+    ad renders.

```

Comment:
When do the reports actually get sent?

Reply:
Later when the ad renders.  I think this spec just needs to populate the URLs in the FencedFrameConfig, and when the URN gets swapped in for the URL, then they can be fetched.  I imagine this can be done by the Fenced Frame spec.